core/frontend/src/tools/Tool.ts

```diff
@@ -646,11 +646,14 @@ export abstract class InteractiveTool extends Tool {
       toolAdmin.setLocateCursor(enableLocate);
     }
 
-    accuSnap.enableLocate(enableLocate);
-    if (undefined !== enableSnap)
-      accuSnap.enableSnap(enableSnap);
-    else
-      accuSnap.enableSnap(false);
+    // Always set the one that is true first, otherwise AccuSnap will clear the TouchCursor for no apparent reason. 

```

Comment:
Not going to approve with this "no apparent reason" comment, there's obviously a reason.

Reply:
Right... There is. As described in the conversation, the TouchCursor isn't only invoked by enabling snap anymore. When `ToolSettings.enableVirtualCursorForLocate=true`, enabling locate will invoke it as well. If both locate/snap end up being turned off at the same time, the TouchCursor gets cleared, even if the next statement is to turn one of them back on. I've updated the example in the original post.

Reply:
Right, and it's not true that a tool is always snapping or locating. The select tool for examples, pick a few elements using the touch cursor, then change mode to line or box select...not neither locate or snap is enabled and the touch cursor needs to go away. This change should be fine, any tool that enables locate really needs to call this method as it does a lot of things you'd get wrong otherswise, i.e. cursor, locate circle, coord lock overrides, etc.